### PR TITLE
Lcam 225 check cc details

### PR DESCRIPTION
**Pull request by AafAnsari**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-225)

- Added crown court outcome validation, exception handler, unit tests
- Update json schema with outcome fields

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by venkat980** [comment]
The test coverage can be improved and the code smells need to be addressed.

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidator.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        Optional<ApiCrownCourtOutcome> apiCrownCourtOutcome = ofNullable(crownCourtSummary.getCrownCourtOutcome())
+                .orElse(emptyList()).stream()
+                .filter(outcome -> outcome.getDateSet() == null)
+                .findFirst();
+
+        if (apiCrownCourtOutcome.isPresent()) {
+            final CrownCourtOutcome crownCourtOutcome = apiCrownCourtOutcome.get().getOutcome();
+            log.info("Validating crown court outcome " + crownCourtOutcome);
+            if (crownCourtOutcome != null) {
```
Comment (venkat980) [inline crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidator.java:36]: the two if statements can be merged here.

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    @Test
+    public void givenValidCrownCourtOutcomeWithNullDate_whenValidateIsInvoked_thenValidationPasses() {
+        ApiCrownCourtOutcome crownCourtOutcome = TestModelDataBuilder.getApiCrownCourtOutcome(CrownCourtOutcome.SUCCESSFUL, null);
+        ApiCrownCourtSummary crownCourtSummary = TestModelDataBuilder.getCrownCourtSummaryWithOutcome(null, Arrays.asList(crownCourtOutcome));
+
+        assertThat(crownCourtDetailsValidator.validate(crownCourtSummary)).isEmpty();
+    }
+
+    @Test
+    public void givenCrownCourtOutcomeConvictedWithNullDateAndNullImprisoned_whenValidateIsInvoked_thenValidationFails() {
```
Comment (venkat980) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidatorTest.java:85]: please remove the public modifier from the test methods

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import uk.gov.justice.laa.crime.crowncourt.model.ApiCrownCourtSummary;
+import uk.gov.justice.laa.crime.crowncourt.staticdata.enums.CrownCourtOutcome;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+public class CrownCourtDetailsValidatorTest {
```
Comment (venkat980) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidatorTest.java:16]: please remove this public modifier

**Review by venkat980 — COMMENTED** [review]
The test coverage can be improved.

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidatorTest.java (hunk 2 above)
Comment (AafAnsari) [reply crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/validation/CrownCourtDetailsValidatorTest.java:85]: an oversight. I'll remove these

**Comment by AafAnsari** [comment]
> 

I have 100% code coverage for the validator class

**Comment by AafAnsari** [comment]
> > > 
> > 
> > 
> > I have 100% code coverage for the validator class
> 
> if (apiCrownCourtOutcome.isPresent() && apiCrownCourtOutcome.get().getOutcome() != null) { One more condition needs to be covered in the Validator class. Can you please check?

I have removed this check and added the outcome field as required in the schema

**Review by venkat980 — APPROVED** [review]

**Pull request merged**